src/ValueObject/GalleryValueObject.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+/**
+ * Handle information about a media item.
+ */
+class GalleryValueObject extends ValueObjectBase {
+
+  /**
+   * The icon of the gallery item.
+   *
+   * @var string
+   */
+  protected $icon;
+
+  /**
+   * The caption of the gallery item.
+   *
+   * @var string
+   */
+  protected $caption;
+
+  /**
+   * Extra classes of the gallery item.
+   *
+   * @var string
+   */
+  protected $classes;
+
+  /**
+   * Extra classes of the gallery item.
+   *
+   * @var \Drupal\oe_theme\ValueObject\ImageMediaValueObject
+   */
+  protected $image;
+
+  /**
+   * GalleryType constructor.
+   *
+   * @param \Drupal\oe_theme\ValueObject\ImageMediaValueObject $image
+   *   Image to be rendered on the gallery item.
+   * @param string|null $icon
+   *   Icon for the gallery item.
+   * @param string|null $caption
+   *   Caption for the gallery item.
+   * @param string|null $classes
+   *   Extra classes for the gallery item.
+   */
+  public function __construct(ImageMediaValueObject $image, string $icon = NULL, string $caption = NULL, string $classes = NULL) {
+    $this->icon = $icon;
```

Comment:
I would derive the icon from the type of media, we can enclose this business logic within this media object so users will not have to worry about deciding which icon is appropriate.

Reply:
The icon is not connected to the type of media. Because the gallery is always an image and the link should be a video, an image, an audio or a text. So the icon is absolutely connected to the gallery item.

Reply:
Agree with @ademarco.